app/models/messages.py

```diff
@@ -0,0 +1,117 @@
+from flask import jsonify
+from app import db
+
+def success_message(db_item, status_code):
+
+    class_name = db_item.__class__.__name__.lower()
+
+    if class_name=="task" and db_item.goal_id:
+        return jsonify({
+            class_name : db_item.to_dict_with_relationship()
+        }), status_code
+    
+    return jsonify({
+            class_name : db_item.to_dict()
+        }), status_code
+
+def invalid_data_message():
+    return jsonify({ "details" : "Invalid data" }), 400
+
+def validate_id(Item, id):
+
+    try:
+        int(id) == id
+
+    except ValueError:
+        return invalid_data_message()
+
+    item = Item.query.get(id)
+
+    if not item:
+        return "", 404
```

Comment:
Consider adding Flask `abort` to simplify some of the code around this function.  Flask `abort` makes it possible to bypass the calling function and return the 400 or 404 messages directly out of a helper.  
```suggestion
        abort(404)
```
This will generate an HTML message by default, to return json make_response can be used with abort, ex: `abort(make_response("", 404))`